crates/storage/provider/src/providers/database/mod.rs

```diff
@@ -46,6 +46,8 @@ pub struct ProviderFactory<DB> {
     chain_spec: Arc<ChainSpec>,
     /// Static File Provider
     static_file_provider: StaticFileProvider,
+    /// Optional pruning configuration
+    prune_modes: Option<PruneModes>,
```

Comment:
does this even have to be an Option?

all the fields of this types are options

Reply:
hmmm, I guess it doesn't need to, we can just use `Default` for the `None` case